api/v1alpha1/profile_types.go

```diff
@@ -28,17 +37,23 @@ type ProfileDefinitionSpec struct {
 	// Description is some text to allow a user to identify what this profile installs.
 	Description string `json:"description,omitempty"`
 	// Artifacts is a list of Profile artifacts
-	// can be one of HelmChart, TODO
 	Artifacts []Artifact `json:"artifacts,omitempty"`
 }
 
+// Artifact defines a bundled resource of the components for this profile.
 type Artifact struct {
 	// Name is the name of the Artifact
 	Name string `json:"name,omitempty"`
 	// Path is the local path to the Artifact in the Profile repo
 	Path string `json:"path,omitempty"`
-	// Kind is the kind of artifact: HelmChart or Kustomize
+	// Kind is the kind of artifact: HelmChartLocal or Kustomize
 	Kind string `json:"kind,omitempty"`
```

Comment:
We could add a kubebuilder validation annotation for the enumeration here?

Reply:
Absolutely 👍

Reply:
Added.

Reply:
I was thinking more of something like:
```
// +kubebuilder:validation:Enum=HelmChartLocal;Kustomize
```

Which would embed this into the CRD (and validate).

But this also hints at a possible smell `HelmChartLocal` and `Kustomize` don't feel at the same level of abstraction?

Reply:
I think you are looking at an outdated revision. I think this is github messing up the resolved comment.
I completely removed the notion of `Local` and now I'm just using the optional Chart value and Path value. :)